.tekton/basic-component-pipeline.yaml

```diff
@@ -356,6 +356,30 @@ spec:
       operator: in
       values: [ "false" ]
 
+  - name: roxctl-image-scan
+    params:
+    - name: image
+      value: $(tasks.build-container.results.IMAGE_URL)@$(tasks.build-container.results.IMAGE_DIGEST)
+    - name: output_format
+      value: table
+    - name: rox_image
+      value: quay.io/stackrox-io/roxctl:4.4.2
```

Comment:
1. One idea is to introduce a floating tag like `:4` similar to how many GitHub actions handle this.
2. If we can provide a pull secret we can switch to downstream, yes.
3. Via a secret in the application workspace. Although that will change with the next version of the tasks.

Reply:
Re 2: we already have the secret in Konflux (see [this doc](https://spaces.redhat.com/pages/viewpage.action?pageId=407312060#id-[DRAFT]HowtoeverythingKonflux/RHTAPforRHACS-WhatsecretsdowehaveinKonflux)) and it should be effective. Worth giving a try.

Re 1: We don't have `:4` or `:latest` floating tags downstream and at some point we decided it's anti-pattern. `roxctl` is a bit special in this regard so we could consider doing that if it's possible to configure for `roxctl` alone but not for other containers (I don't remember).
Konflux has renovate bot which sends us PRs like <https://github.com/stackrox/stackrox/pull/11091>. It's set up to work against `master` branch. Its config seems to be <https://github.com/konflux-ci/build-service/blob/main/controllers/git_tekton_resources_renovater.go>. Maybe we'll get that automatically?

Re 3: secret for central URL feels unnatural. I assume you have created one. If so, please add to the [Confluence table](https://spaces.redhat.com/pages/viewpage.action?pageId=407312060#id-[DRAFT]HowtoeverythingKonflux/RHTAPforRHACS-WhatsecretsdowehaveinKonflux).

Reply:
Also, regarding 1: Could you please brief me why reading version from Central to determine roxctl version is not good for image tag?

Reply:
Continuing https://github.com/stackrox/stackrox/pull/11192#issuecomment-2145073697 here.

1.1.

> We decided on supplying roxctl via image for two reasons.

Even when using the image, you can figure out the image tag from Central metadata, or not?

1.2.

> That's how the existing Tekton tasks worked.

Not changing something because that's how it works is an argument that's hard to beat.
Assuming for a second that you'd be open to change the Task, would you agree that having `roxctl` and Central versions aligned is beneficial?

1.3.

> Contrary to Github actions, Tekton tasks may be run with read-only filesystems. This can cause problems with donwloading roxctl from Central.

You can define an `emptyDir` volume in the Task definition and use it for writing. https://tekton.dev/docs/pipelines/pipeline-api/#tekton.dev/v1.Task